pkg/cmd/adm/register_member.go

```diff
@@ -351,35 +488,20 @@ func (v *registerMemberValidated) confirmationPrompt() ioutils.ConfirmationMessa
 	return ioutils.WithMessagef(sb.String(), args...)
 }
 
-func (v *registerMemberValidated) perform(ctx *extendedCommandContext, newCommand client.CommandCreator) error {
+func (v *registerMemberValidated) perform(ctx *extendedCommandContext) error {
 	// add the host entry to the member cluster first. We assume that there is just 1 toolchain cluster entry in the member
 	// cluster (i.e. it just points back to the host), so there's no need to determine the number of entries with the same
 	// API endpoint.
-	hostToolchainClusterKey := runtimeclient.ObjectKey{
-		Name:      v.hostToolchainClusterName,
-		Namespace: v.args.memberNamespace,
-	}
-	if err := runAddClusterScript(ctx, newCommand, configuration.Host, v.args.hostKubeConfig, v.args.hostNamespace, v.args.memberKubeConfig, v.args.memberNamespace, "", v.args.useLetsEncrypt); err != nil {
-		return err
-	}
-
-	if err := waitUntilToolchainClusterReady(ctx.CommandContext, v.memberClusterClient, hostToolchainClusterKey, v.waitForReadyTimeout); err != nil {
-		ctx.Println("The ToolchainCluster resource representing the host in the member cluster has not become ready.")
-		ctx.Printlnf("Please check the %s ToolchainCluster resource in the %s member cluster.", hostToolchainClusterKey, v.memberApiEndpoint)
+	if err := v.addCluster(ctx, configuration.Host); err != nil {
 		return err
 	}
 
+	// add the member entry in the host cluster
 	memberToolchainClusterKey := runtimeclient.ObjectKey{
-		Name:      v.memberToolchainClusterName,
-		Namespace: v.args.hostNamespace,
+		Name:      v.memberClusterData.toolchainClusterName,
+		Namespace: v.hostClusterData.namespace,
 	}
```

Comment:
minor - you can drop this object and set the fields in the SPC directly:
```go
exampleSPC := &toolchainv1alpha1.SpaceProvisionerConfig{
	TypeMeta: metav1.TypeMeta{
		Kind:       "SpaceProvisionerConfig",
		APIVersion: toolchainv1alpha1.GroupVersion.Identifier(),
	},
	ObjectMeta: metav1.ObjectMeta{
		Name:      v.memberClusterData.toolchainClusterName,
		Namespace: v.hostClusterData.namespace,
	},
	Spec: toolchainv1alpha1.SpaceProvisionerConfigSpec{
		ToolchainCluster: v.memberClusterData.toolchainClusterName,
		Enabled:          false,
		PlacementRoles: []string{
			cluster.RoleLabel(cluster.Tenant),
		},
	},
}
```

Reply:
good catch, it was a leftover. Done in https://github.com/kubesaw/ksctl/pull/53/commits/9471f72065ea905c555c1bcd12977c075cdb82e6